### ISSUE TITLE
Quesiton about the implementation

### ISSUE THREAD

**Issue by hiker-lw** [AttendAndExcite/Attend-and-Excite#49]
Hi, thank you for your excellent work! I have a few questions regarding the code implementation:

1. Regarding the normalization of Cross-Attention Maps: In the code `attention_for_text = attention_maps[:, :, 1:last_idx]`, it appears that the last `<eot>` token is also excluded after excluding `<sot>`. May I ask why this is done?
2. Regarding `perform_iterative_refinement_step`: It seems that the following two forward passes might be unnecessary, if I understand correctly.

```
with torch.no_grad():
        noise_pred_uncond = self.unet(latents, t, encoder_hidden_states=text_embeddings[0].unsqueeze(0)).sample
        noise_pred_text = self.unet(latents, t, encoder_hidden_states=text_embeddings[1].unsqueeze(0)).sample
```
Looking forward to your response! Thanks.
